### PR TITLE
PIN-3558: added AOO and UO in loadOpenData

### PR DESCRIPTION
**Pull request by AsterITA**

### CODE REVIEW

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -44,6 +44,12 @@ party-registry-proxy {
       institutions {
         open-data-url = ${PARTY_REGISTRY_INSTITUTIONS_URL}
       }
+      AOO {
+        open-data-url = ${PARTY_REGISTRY_AOO_URL}
+      }
+      UO {
```
Comment (galales) [inline src/main/resources/application.conf]: very minor naming convention
```suggestion
      aoo {
        open-data-url = ${PARTY_REGISTRY_AOO_URL}
      }
      uo {
```

File: src/main/scala/it/pagopa/interop/partyregistryproxy/common/system/ApplicationConfiguration.scala [hunk 2]
```diff
@@ -19,6 +19,12 @@ object ApplicationConfiguration {
   val institutionsIpaOpenDataUrl: String =
     config.getString("party-registry-proxy.sources.ipa.institutions.open-data-url")
 
+  val AOOIpaOpenDataUrl: String =
+    config.getString("party-registry-proxy.sources.ipa.AOO.open-data-url")
+
+  val UOIpaOpenDataUrl: String =
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/partyregistryproxy/common/system/ApplicationConfiguration.scala]: same
```suggestion
  val aooIpaOpenDataUrl: String =
    config.getString("party-registry-proxy.sources.ipa.AOO.open-data-url")

  val uoIpaOpenDataUrl: String =
```

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala [hunk 3]
```diff
@@ -6,22 +6,31 @@ import akka.http.scaladsl.model.HttpRequest
...
+    categorySource: Map[String, String],
+    institutionKind: InstitutionKind
+  ): Future[List[Institution]] = {
+    val url = institutionKind match {
+      case InstitutionKind.Agency => ApplicationConfiguration.institutionsIpaOpenDataUrl
+      case InstitutionKind.AOO    => ApplicationConfiguration.AOOIpaOpenDataUrl
+      case InstitutionKind.UO     => ApplicationConfiguration.UOIpaOpenDataUrl
+    }
+
+    retrieveOpenData(url).map(IPAOpenDataServiceImpl.extractInstitutions(categorySource, institutionKind))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala]:
```suggestion
    retrieveOpenData(url).map(extractInstitutions(categorySource, institutionKind))
```

File: src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala [hunk 4]
```diff
@@ -69,9 +82,19 @@ object IPAOpenDataServiceImpl {
...
+          case InstitutionKind.AOO    => record(mapped(InstitutionsFields.aooId)).select[String]
+          case InstitutionKind.UO     => record(mapped(InstitutionsFields.uoId)).select[String]
+        }
         taxCode        <- mapped.get(InstitutionsFields.taxCode).flatMap(idx => record(idx).select[String])
-        category       <- mapped.get(InstitutionsFields.category).flatMap(idx => record(idx).select[String])
+        category       <- institutionKind match {
+          case InstitutionKind.Agency =>
+            mapped.get(InstitutionsFields.category).flatMap(idx => record(idx).select[String])
+          case _                      =>
+            val institutionOriginId = record(mapped(InstitutionsFields.originId)).select[String]
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala]: Despite it should not happen, this is safer
```suggestion
            val institutionOriginId = mapped.get(InstitutionsFields.originId).flatMap(idx => record(idx).select[String])
```

**Review by galales — COMMENTED** [review]
Should [these](https://github.com/pagopa/interop-be-party-registry-proxy/pull/153/files#diff-5b482c43116481f280b50a1fb597b3baddda08e51ca749292dfb8efeda7cdd24R107-R108) be updated?

**Comment by AsterITA** [comment]
> Should [these](https://github.com/pagopa/interop-be-party-registry-proxy/pull/153/files#diff-5b482c43116481f280b50a1fb597b3baddda08e51ca749292dfb8efeda7cdd24R107-R108) be updated?

I think you are right, but it's not clear to me what these fields should contain.

Should it be like following?

if AOO:
`o = None,
ou = None,
aoo = aooId`

if UO:
`o = None,
ou = uoID,
aoo = None`?

In the code both `uoID` and `aooId` are `originId`.

**Comment by galales** [comment]
TBH I'm not sure. Also, based on the comment in the `TODO`, probably the id is the one in selfcare.
@beetlecrunch is THE ONE who can dispel our doubts 🪄

**Comment by beetlecrunch** [comment]
> TBH I'm not sure. Also, based on the comment in the `TODO`, probably the id is the one in selfcare. @beetlecrunch is THE ONE who can dispel our doubts 🪄

Those fields (o, uo, aoo) come from a legacy logic that has never been applied.
They can be removed from the `Institution` model because they are not used anywhere in our code base.
Removing them means verifying compilation and tests in every service that uses the party-registry proxy.

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**